Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1273,6 +1273,15 @@ void FrameLoader::loadInSameDocument(URL url, RefPtr<SerializedScriptValue> stat
 
     checkedHistory()->updateForSameDocumentNavigation();
 
+    if (document->settings().navigationAPIEnabled() && !equalIgnoringFragmentIdentifier(url, aboutBlankURL())) {
```

Comment:
`URL.isAboutBlank()` ignores the fragment.

Reply:
Nice! Fixed.